src/build/patches.ts

```diff
@@ -135,13 +160,58 @@ function handleEvent(child: Node): Event {
  */
 function handleProperty(child: Node): Partial<Property> {
   return {
-    name: child.values[0] as string,
+    name: string(child.values[0]),
     ...optionalMember("exposed", "string", child.properties?.exposed),
     ...optionalMember("optional", "boolean", child.properties?.optional),
     ...optionalMember("overrideType", "string", child.properties?.overrideType),
   };
 }
 
+/**
+ * Handles a child node of type "method" and adds it to the method object.
+ * @param child The child node to handle.
+ */
+function handleMethod(child: Node): Partial<Method> {
+  const name = string(child.values[0]);
+  const returnType = child.properties.returns;
+
+  let typeNode: Node | undefined;
+  const params: { name: string; type: string }[] = [];
+
+  for (const c of child.children) {
+    switch (c.name) {
+      case "type":
+        if (typeNode) {
+          throw new Error(`Method "${name}" has multiple type nodes (invalid)`);
+        }
+        typeNode = c;
+        break;
+
+      case "param":
+        params.push({
+          name: string(c.values[0]),
+          type: string(c.properties.type),
+        });
+        break;
+
+      default:
+        throw new Error(`Unexpected child "${c.name}" in method "${name}"`);
+    }
+  }
+
+  if (!typeNode) {
+    throw new Error(`Method "${name}" is missing a return type`);
+  }
+
+  const signature: Method["signature"] = [
+    {
+      param: params,
+      ...handleTyped(typeNode, returnType),
+    },
+  ];
+  return { name, signature };
```

Comment:
(This is fine for now but eventually we'll hit a problem when dealing with overloads)